test/test-util.ts

```diff
@@ -97,8 +97,86 @@ test('should follow dependency chain caused by extends files', async t => {
 });
 
 
+test('should throw an error if it finds a circular reference', async t => {
+  const FAKE_DIRECTORY = '/some/fake/directory';
+  const FAKE_CONFIG1 = {files: ['b'], extends: 'FAKE_CONFIG2'};
+  const FAKE_CONFIG2 = {extends: 'FAKE_CONFIG3'};
+  const FAKE_CONFIG3 = {extends: 'FAKE_CONFIG1'};
+  function fakeReadFilep(
+      configPath: string, encoding: string): Promise<string> {
+    switch (configPath) {
+      case '/some/fake/directory/FAKE_CONFIG1':
+
+      case '/some/fake/directory/FAKE_CONFIG2':
+        return Promise.resolve(JSON.stringify(FAKE_CONFIG2));
+      case '/some/fake/directory/FAKE_CONFIG3':
+        return Promise.resolve(JSON.stringify(FAKE_CONFIG3));
+      case '/some/fake/directory/tsconfig.json':
+        return Promise.resolve(JSON.stringify(FAKE_CONFIG1));
+      default:
+        return Promise.reject('File Not Found');
+    }
+  }
+  await t.throws(
+      getTSConfig(FAKE_DIRECTORY, fakeReadFilep), Error,
+      'Circular Reference Detected');
+});
+
+test(
+    'when a file contains an extends field, the base file is loaded first then overridden by the inherited files',
```

Comment:
This seems like it is already covered in a test above. (`should follow dependency chain caused by extends files`)

Reply:
I see, makes sense.